src/compiler/lib/src/jsMain/kotlin/community/flock/wirespec/compiler/lib/Results.kt

```diff
@@ -35,3 +35,15 @@ class WsParseResult(
     val result: Array<WsNode>? = null,
     val errors: Array<WsError>? = null,
 )
+
+fun Either<Nel<WirespecException>, String>.produce(): WsStringResult = when (this) {
+    is Either.Left -> WsStringResult(errors = value.map { it.produce() }.toTypedArray())
+    is Either.Right -> WsStringResult(result = value)
+}
+
+@JsExport
+class WsStringResult(
```

Comment:
Why is WsResult not enough?

Reply:
The content is string ?

Reply:
@JsExport
class WsCompilationResult(
    val result: WsCompiled? = null,
    val errors: Array<WsError> = emptyArray()
)

@JsExport
class WsCompiled(val value: String)

String result is not needed for WsCompiled is a string